tools/management/commands/get_gtp_ligands.py

```diff
@@ -0,0 +1,493 @@
+                    context[name]['pec50_min'].append(assay['pec50_min'])
+                if assay['pec50_max'] is not None and isinstance(assay['pec50_max'], float):
+                    context[name]['pec50_max'].append(assay['pec50_max'])
+                if assay['pKi_avg'] is not None and isinstance(assay['pKi_avg'], float):
+                    context[name]['pKi_avg'].append(assay['pKi_avg'])
+                if assay['pKi_min'] is not None and isinstance(assay['pKi_min'], float):
+                    context[name]['pKi_min'].append(assay['pKi_min'])
+                if assay['pKi_max'] is not None and isinstance(assay['pKi_max'], float):
+                    context[name]['pKi_max'].append(assay['pKi_max'])
+                context[name]['references'].extend(assay['references'])
+            else:
+                context[name] = dict()
+
+                context[name]['ligand'] = assay['ligand']
+                context[name]['ligand_type'] = assay['ligand_type']
+                context[name]['endogenous_princip'] = assay['endogenous_princip']
+                context[name]['receptor'] = assay['receptor']
+
+                context[name]['pec50_avg'] = list()
+                context[name]['pec50_min'] = list()
+                context[name]['pec50_max'] = list()
+                if assay['pec50_avg'] is not None and isinstance(assay['pec50_avg'], float):
+                    context[name]['pec50_avg'].append(assay['pec50_avg'])
+                if assay['pec50_min'] is not None and isinstance(assay['pec50_min'], float):
+                    context[name]['pec50_min'].append(assay['pec50_min'])
+                if assay['pec50_max'] is not None and isinstance(assay['pec50_max'], float):
+                    context[name]['pec50_max'].append(assay['pec50_max'])
+
+                context[name]['pKi_avg'] = list()
+                context[name]['pKi_min'] = list()
+                context[name]['pKi_max'] = list()
+
+                if assay['pKi_avg'] is not None and isinstance(assay['pKi_avg'], float):
+                    context[name]['pKi_avg'].append(assay['pKi_avg'])
+                if assay['pKi_min'] is not None and isinstance(assay['pKi_min'], float):
+                    context[name]['pKi_min'].append(assay['pKi_min'])
+                if assay['pKi_max'] is not None and isinstance(assay['pKi_max'], float):
+                    context[name]['pKi_max'].append(assay['pKi_max'])
+
+                context[name]['gpt_link'] = assay['gpt_link']
+                context[name]['references'] = list()
+                context[name]['references'].extend(assay['references'])
+        return context
+
+    def prepare_calculate_averages_and_save(self, context):
+        prepared_data = list()
+        for i in context.items():
+            prepared_data.append(i[1])
+        return prepared_data
+
+    def calculate_averages(self, prepared_data):
+        for assay in prepared_data:
+
+            try:
```

Comment:
Can these try/except be merged into a for cycle that cycles through the keys of 'assay'?

Reply:
as for these, it is better to remain as it is because in most cases some of assay[xyz] will be empty, but we still need to calculate the median value to assign to gpcrdb value